abd-clam/src/cakes/knn/sieve_v1.rs

```diff
@@ -291,13 +197,13 @@ impl<'a, T: Send + Sync + Copy, U: Number> Grain<'a, T, U> {
     /// Changes pivot point and swaps elements around so that all elements to left
     /// of pivot are less than or equal to pivot and all elements to right of pivot
     /// are greater than pivot.
-    pub fn _partition(grains: &mut [Self], l: usize, r: usize) -> usize {
-        let pivot = (l + r) / 2;
+    fn _partition_once(grains: &mut [Self], l: usize, r: usize) -> usize {
+        let pivot = l + (r - l) / 2;
         grains.swap(pivot, r);
 
         let (mut a, mut b) = (l, l);
         while b < r {
-            if grains[b].d <= grains[r].d {
+            if grains[b].d < grains[r].d {
```

Comment:
I forgot about this until now but my original reason for having it this way is that it allows the index returned from partition_kth to eliminate the need for the is_outside() function in sieve V1. You just only take grains[0:i] and you have everything you need and don't need to check one by one if they're outside or not. But if you get rid of the = you get less swaps which is good but you forego the ability to eliminate a bunch of grains without iterating over all of them. Not sure if that's worth considering. Either way, if you're going to get rid of the = you should change the documentation on this function because it is no longer correct.